akka-stream-tests/src/test/scala/akka/stream/scaladsl/FlowConcatAllSpec.scala

```diff
@@ -44,7 +44,13 @@ class FlowConcatAllSpec extends AkkaSpec {
 
     "work together with SplitWhen" in {
       val subscriber = TestSubscriber.manualProbe[Int]()
-      Source(1 to 10).splitWhen(_ % 2 == 0).flatMapConcat(ConstantFun.scalaIdentityFunction).runWith(Sink(subscriber))
+      Source(1 to 10)
+        .splitWhen(_ % 2 == 0)
+        .prefixAndTail(0)
+        .map(_._2)
+        .concatSubstreams
+        .flatMapConcat(ConstantFun.scalaIdentityFunction)
```

Comment:
```
.prefixAndTail(0)
.map(_._2)
.concatSubstreams
```
is a funky way to way to lift the substreams to actual elements of the stream. It is funky because it is dangerous therefore we don't provide sugar for it. The additional `flatMapConcat` is part of the integration test which is to see if splitWhen and concatAll works together.